### DISCUSSION TITLE
Typescript error on components using tabs and tabsitem with Docusaurus 2.3

### DISCUSSION THREAD

**Discussion by adriantam** [facebook/docusaurus#8643] [Q&A]
Anyone having trouble using tabs in their React component with Docusaurus 2.3?  Typescript is complaining about type `{ children: Element; value: string; label: string; }` is not assignable to type 'IntrinsicAttributes & Props'.

For your information, my code looks something like this

```
export const Component = (): JSX.Element => {
  return (
    <>
    <Tabs groupId="abc">
      <TabItem value="1" key="a">
        XYZ
      </TabItem>
    </Tabs>
    </>
   );
} 
```

The exact error is

```
src/components/Docs/AuthorizationModel/AuthzModelSnippetViewer.tsx:35:8 - error TS2322: Type '{ children: Element[]; groupId: string; }' is not assignable to type 'IntrinsicAttributes & Props'.
  Property 'children' does not exist on type 'IntrinsicAttributes & Props'.

35       <Tabs groupId="dsl">
          ~~~~

src/components/Docs/AuthorizationModel/AuthzModelSnippetViewer.tsx:36:10 - error TS2322: Type '{ children: Element; value: string; label: string; }' is not assignable to type 'IntrinsicAttributes & Props'.
  Property 'children' does not exist on type 'IntrinsicAttributes & Props'.

36         <TabItem value="dsl" label="DSL">
            ~~~~~~~
```

This was not the issue in Docusaurus 2.2.  Note that building the component is successful, however, typescript (i.e., tsc  --skipLibCheck) is asserting error. 

if we update the theme-class.d.ts such that 

```
declare module '@theme/TabItem' {
  import type {TabItemProps} from '@docusaurus/theme-common/internal';

  export interface Props extends TabItemProps {}

  export default function TabItem(props: TabItemProps): JSX.Element;
}

declare module '@theme/Tabs' {
  import type {TabsProps} from '@docusaurus/theme-common/internal';

  export interface Props extends TabsProps {}

  export default function Tabs(props: TabsProps): JSX.Element;
}
```

instead of 
```
declare module '@theme/TabItem' {
  import type {TabItemProps} from '@docusaurus/theme-common/internal';

  export interface Props extends TabItemProps {}

  export default function TabItem(props: Props): JSX.Element;
}

declare module '@theme/Tabs' {
  import type {TabsProps} from '@docusaurus/theme-common/internal';

  export interface Props extends TabsProps {}

  export default function Tabs(props: Props): JSX.Element;
}
```
, 
we don't see the typescript error anymore.  It appears that it is having problem understanding the extended interface.

**Comment by slorber** [comment]
This looks surprising to me, I don't see much difference in your TS diff that would explain why it pass or fail. 

If you still have the issue please create a small repro initializing a new Docusaurus + TS project.

**Reply by adriantam** [reply]
@slorber : Thank you for your reply.  I have created a very simple repo that reproduces the issue.  Please see https://github.com/adriantam/docusaurus_tab.

The only change from `npx create-docusaurus@latest my-website classic --typescript` is having the file https://github.com/adriantam/docusaurus_tab/blob/main/src/components/TabEx/TabEx.tsx

If we downgrade to Docusaurus 2.2.0, running  `npm run typecheck` passes with no problem.

This is problematic with Docusaurus 2.3.0 and 2.3.1.

**Reply by slorber** [reply]
🤷‍♂️ can reproduce but have no idea what is happening here.

Looks related to changes made in https://github.com/facebook/docusaurus/pull/8486

@Josh-Cena any idea?

Can't reproduce the issue using the same code in our own website, but I do in a newly initialized project.

**Reply by slorber** [reply]
Oh I see, it's because of using package exports, same issue as https://github.com/facebook/docusaurus/issues/8226

Try adding `"moduleResolution": "Node16"` to your TS config

**Reply by adriantam** [reply]
Yes, this solves the issue.  Thank you very much for your help!
